### PR TITLE
Backport changes from release/2.9

### PR DESCRIPTION
**Pull request by StephenSorriaux**
## Why is this needed?
- bump the version
- update changelog
- backport the fix that overrides the `egg_info` `tag-build` configuration

## Proposed Changes

  - bump version to 2.9.0
  - add changelog for 2.9.0
  - add some flags to `build` module to override `tag-build` value from `setup.cfg`

## Does this PR introduce any breaking change?
No

### CODE REVIEW

**Review by ceache — APPROVED** [review]

File: kazoo/version.py [hunk 1]
```diff
@@ -1 +1 @@
-__version__ = '2.8.0'
```
Comment (ceache) [inline kazoo/version.py:1]: FYI black, in my PR, will complain about the single quotes.

On a separate note, I think we should move to setuptools_scm and have this file generated for us from git tags, what do you think?

Comment (StephenSorriaux) [reply kazoo/version.py:1]: Love the idea!

**Comment by StephenSorriaux** [comment]
Mh I made a mistake by merging https://github.com/python-zk/kazoo/pull/671 before, because now I don't want to force push to the `release/2.9` branch to get those changes back. Will open another PR and cherry-pick.

File: kazoo/version.py (hunk 1 above)
Comment (ceache) [reply kazoo/version.py:1]: I have it ready to go, I'll submit a PR tomorrow. But I'd like it on top of my black PR if possible

**Pull request closed without merging**